File: 2025-06-June-GFG-POTD/June 28 - Counting elements in two arrays/Solution.py
from bisect import bisect_right

class Solution:
    @staticmethod
    def countLessEq(a, b):
        result = []
        
        # Step 1: Sort array b
        b.sort()
        
        # Step 2: For each element in a, use bisect_right
        for num in a:
            count = bisect_right(b, num)
            result.append(count)
        
        return result

File: 2025-06-June-GFG-POTD/June 28 - Counting elements in two arrays/Solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    static vector<int> countLessEq(vector<int>& a, vector<int>& b) {
        vector<int> result;

        // Step 1: Sort array b
        sort(b.begin(), b.end());

        // Step 2: For each element in a, perform upper_bound search in b
        for (int num : a) {
            int count = upperBound(b, num);
            result.push_back(count);
        }

        return result;
    }

private:
    // Function to find count of elements <= x using binary search
    static int upperBound(const vector<int>& arr, int x) {
        int low = 0, high = arr.size();

        while (low < high) {
            int mid = (low + high) / 2;
            if (arr[mid] <= x) {
                low = mid + 1;
            } else {
                high = mid;
            }
        }

        return low; // number of elements <= x
    }
};
